nixos/modules/tasks/network-interfaces.nix

```diff
@@ -521,7 +521,10 @@ in
     };
 
     networking.hostId = mkOption {
-      default = null;
+      default = lib.optionalString (config.networking.machineId != null) lib.substring 0 8 config.networking.machineId;
```

Comment:
I just notice that when the predicate is `false`, the default value will be an empty string. Is it OK ?

Reply:
Also it doesn't match the `defaultText` so those should be reconciled either way.

Reply:
And looking at the `mkIf`s later on, no, I don't think it's okay for it to be an empty string.

Reply:
You're right, it should be null or nonEmpty – I added a suggestion for the `type` to reflect that.

Reply:
It should be resolved now, please check if I understood correctly :)